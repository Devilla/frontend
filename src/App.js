import React from 'react';
import { browserHistory } from 'react-router';
import { AppContainer } from 'react-hot-loader';
import { createStore, applyMiddleware } from 'redux';
import { Provider } from 'react-redux';
import rootSaga from './sagas';
import reducer from './ducks';
import thunk from 'redux-thunk';
import { syncHistoryWithStore } from 'react-router-redux';
import { composeWithDevTools } from 'redux-devtools-extension/developmentOnly';
import createSagaMiddleware from 'redux-saga';
import { StripeProvider } from 'react-stripe-elements';

const sagaMiddleware = createSagaMiddleware();

export const store = createStore(reducer, composeWithDevTools(
  applyMiddleware(sagaMiddleware, thunk)
));

const routerHistory = syncHistoryWithStore(browserHistory, store, {selectLocationState: state => state.get('router')});


sagaMiddleware.run(rootSaga);

const App = ({Component}) =>
  <AppContainer errorReporter={Error}>
    <Provider store={store}>
      <StripeProvider apiKey={process.env.NODE_ENV === 'production' ? process.env.REACT_APP_STRIPE_KEY : process.env.REACT_APP_DEVELOPMENT_STRIPE_KEY}>
        <Component routerHistory={routerHistory} />
      </StripeProvider>
    </Provider>
  </AppContainer>;

export default App;
