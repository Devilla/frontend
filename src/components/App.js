import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import  '../assets/stylesheets/slick.css'
import  '../assets/stylesheets/slick-theme.css'
import  '../assets/stylesheets/style.css'

import Header from './_common/Header';
import Footer from './_common/Footer';
import { Spinner } from './index.js';

class App extends Component {
  render(){
    return (
        <div className="app-container">
          <Header />

          <Spinner loading={this.props.loading} />
          <div className="content">
            {this.props.children}
          </div>
          <Footer />
        </div>
    );
  }
};

App.propTypes = {
  children: PropTypes.node
};

const mapStateToProps = state => ({
  loading: state.getIn(['loading', 'state'])
});

export default connect(mapStateToProps)(App);
