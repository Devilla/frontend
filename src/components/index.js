export { default as WebsiteHeader } from "./WebsiteHeader/WebsiteHeader";
export { default as WebsiteFooter } from "./WebsiteFooter/WebsiteFooter";
export { default as WebsiteHome } from "./WebsiteHome/WebsiteHome";
export { default as WebsiteHowItWorks } from "./WebsiteHowItWorks/WebsiteHowItWorks";
export { default as WebsiteIntegrations } from "./WebsiteIntegrations/WebsiteIntegrations";
export { default as WebsitePricing } from "./WebsitePricing/WebsitePricing";
export { default as WebsiteAbout } from "./WebsiteAbout/WebsiteAbout";
export { default as WebsiteSignIn } from "./WebsiteSignIn/WebsiteSignIn";
export { default as WebsiteSignUp } from "./WebsiteSignUp/WebsiteSignUp";
export { default as WebsiteContact } from "./WebsiteContact/WebsiteContact";
export { default as WebsiteTerms } from "./WebsiteTerms/WebsiteTerms";
export { default as WebsitePolicy } from "./WebsitePolicy/WebsitePolicy";
export { default as WebsiteDemoPage}  from "./WebsiteDemoPage/WebsiteDemoPage";
export { default as WebsiteCustStory}  from "./WebsiteCustStory/WebsiteCustStory";
export { default as WebsiteFeature}  from "./WebsiteFeature/WebsiteFeature";
export { default as Affiliate}  from "./Affiliate/Affiliate";
export { default as AffiliateRegister}  from "./Affiliate/AffiliateRegister";

export { default as Header } from "./Header/Header";
export { default as ForgetPassword } from "./ForgetPassword";
export { default as PricePage } from "./PricePage";
export { default as Common } from "./Common/function";
export { default as ResetPassword } from "./ResetPassword";
export { default as Sidebar } from "./Sidebar/Sidebar";
export { default as StatsCard } from "./StatsCard/StatsCard";
export { default as Dashboard } from './Dashboard/Dashboard';
export { default as Notification } from './Notification';
export { default as Analytics } from './Analytics/analytics';
export { default as AnalyticsProfile } from './Analytics/AnalyticsProfile';
export { default as New } from './New/new';
export { default as LoginFlow } from './LoginFlow/LoginFlow';
export { default as Spinner } from './Spinner/Spinner';
export { default as Error } from './Error/Error';
export { default as ComingSoon } from './ComingSoon/ComingSoon';
export { default as Button } from './Button/Button';
export { default as Profile } from './Profile/Profile';
export { default as BillingDetails } from './Billing/BillingDetails';
export { default as UpgradeCard } from './UpgradeCard/UpgradeCard';
export { default as UpgradePlan } from './UpgradePlan/UpgradePlan';
export { default as Integration } from './Integration/Integration';
export { default as Template } from './Template';
export { default as VerificationPage } from './VerificationPage/VerificationPage';
