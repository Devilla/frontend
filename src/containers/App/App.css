@import url('https://fonts.googleapis.com/css?family=Open+Sans:200,300,400,400i,500,600,700');
@import url('https://fonts.googleapis.com/css?family=Source+Sans+Pro:200,300,400,400i,600,700');

body {
  font-family: "Raleway, Source Sans Pro, Verdana, Helvetica", sans-serif;
}
