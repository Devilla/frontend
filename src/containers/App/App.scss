@import url('https://fonts.googleapis.com/css?family=Open+Sans:200,300,400,400i,500,600,700');
@import url('https://fonts.googleapis.com/css?family=Source+Sans+Pro:200,300,400,400i,600,700');

body {
  font-family: "Raleway, Source Sans Pro, Verdana, Helvetica", sans-serif;
  display: block;
  margin: 0px;
}

.website-app {
  .alert-warning {
    background-color: #f2dede;
    border-color: #ebccd1;
    color: #a94442;
  }
  .content {
    padding-top: 100px;
  }
  .toast-style {
    padding: 17px 7px;
    background-color: rgba(0, 0, 0, 0.9) !important;
    color: #fff !important;
    text-align: center !important;
    height: 10px !important;
    border-radius: 5px !important;
    button {
      height: 1.5em;
      color: #fff;
    }
  }

  .transition-wrapper {
    position: relative;
    z-index: 1;
    .transition-item {
      position: fixed;
      top: 0;
      right: 0;
      left: 0;
    }
  }
  .signin-container,
  .authpage,
  .websitepolicy-container,
  .websitehowitworks-container,
  .websitehome-container,
  .WebsiteTerms-container,
  .websitepricing-container,
  .website-payment-container,
  .websiteintegrations-container,
  .demopage-container,
  .website-cust-story,
  .websitecontact-container,
  .websiteabout-container,
  .WebsiteFeature-container,
  .affiliateregister-container
   {
    overflow: hidden;
    box-sizing: border-box;
    padding: 20px;
    min-height: 100vh;
    transition: 500ms ease-out;

    &.transition-leave {
      opacity: 1;
    }

    &.transition-leave.transition-leave-active {
      opacity: 1;
      transform: translate3d(100%, 0, 0);
    }
  }
}
