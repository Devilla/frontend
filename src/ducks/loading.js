import { Map } from 'immutable';

const action = name => `zikher/modals/${name}`;
const LOAD = action('LOAD');
const LOADED = action('LOADED');

export const load = () => ({ type: LOAD });
export const loaded = () => ({ type: LOADED });

const initialState = Map({
  state: false,
});

const loading = (state = initialState, action) => {
  switch (action.type) {
    case LOAD:
      return state.set('state', true);

    case LOADED:
      return state.set('state', false);

    default:
      return state;
  }
};

export default loading;
