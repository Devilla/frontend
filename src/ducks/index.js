import { combineReducers } from 'redux-immutable';
import { routerReducer } from 'react-router-redux'

import profile from './profile';
import auth from './auth';
import plan from './plan';
import payment from './payment';
import notification from './notification';
import campaign from './campaign';
import rules from './rules';
import loading from './loading';
import configuration from './configuration';
import elastic from './elastic';
import pageurl from './pageurl';
import webhooks from './webhooks';
import subcampaign from './subcampaign';

const reducer = combineReducers({
  profile,
  auth,
  plan,
  payment,
  notification,
  campaign,
  rules,
  loading,
  configuration,
  elastic,
  pageurl,
  webhooks,
  subcampaign,
  router: routerReducer
});

export default reducer;
