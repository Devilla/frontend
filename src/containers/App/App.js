import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { ToastContainer } from 'react-toastify';
import { browserHistory } from 'react-router';
import { connect } from 'react-redux';
import { Spinner } from '../../components/index.js';
import { WebsiteHeader, WebsiteFooter } from 'components';

import { checkTokenExists } from 'ducks/auth';

import './scss/stack-interface.scss';
import './scss/socicon.css';
import './scss/iconsmind.scss';
import './scss/bootstrap.scss';
import './scss/theme.scss';
import './scss/custom.scss';
import './scss/font-raleway.scss';
import './scss/flickity.scss';
import './scss/font-sourcesanspro.scss';
import './App.scss';

class App extends Component {
  componentWillMount() {
    this.checkLogin((err) => {
      if(!err) {
        browserHistory.push('/dashboard');
      }
    });
  }

  checkLogin(callback) {
    const cookie = localStorage.getItem('authToken');
    const authToken = cookie
      ? JSON.parse(cookie)
      : null;
    if (authToken) {
      this.props.checkTokenExists(authToken);
      callback()
    } else {
      callback("not logged in")
    }
  }

  render(){
    return (
      <div className="website-app">
        <div className="basic-gradient-light" data-smooth-scroll-offset="77">
           <WebsiteHeader />
           <Spinner loading={this.props.loading} />
           <div className="content">
             {this.props.children}
           </div>
           <WebsiteFooter />
        </div>
        <ToastContainer hideProgressBar={true} />
      </div>
    );
  }

}

const mapDispatchToProps = {
  checkTokenExists
};

export default connect(null, mapDispatchToProps)(App);
