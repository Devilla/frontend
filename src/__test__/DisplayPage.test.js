import React from 'react';
import '../setupTests';
import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer'
import configureStore from 'redux-mock-store'
import { Provider } from 'react-redux'
import {createStore} from 'redux'
import { load, loaded } from 'ducks/loading';
import { DisplayPage } from 'components';
import { store } from 'App';

// Snapshot for Home React Component
describe('>>> DisplayPage --- Snapshot',()=>{
    it('+++capturing Snapshot of DisplayPage', () => {
        const renderedValue =  renderer.create(<DisplayPage store={store} />).toJSON()
        expect(renderedValue).toMatchSnapshot();
    });
});

describe('check for in', ()=> {
  let wrapper;
  beforeEach ( () => {
       wrapper = shallow(<DisplayPage  store={store}/>);
  })

  it('+++ render the DUMB component', () => {
    expect(wrapper.length).toEqual(1);
  });

  it('+++ contains website - error', () => {
    let wrapper = shallow(<DisplayPage  store={store}/>);
    expect(wrapper.find('.website-error').length).toBe(1);
  });

});
