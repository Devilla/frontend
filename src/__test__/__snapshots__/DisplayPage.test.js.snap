// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>> DisplayPage --- Snapshot +++capturing Snapshot of DisplayPage 1`] = `
<div>
  <div
    className="container-fluid"
  >
    <div
      className="tabscontent"
    >
      <div
        className="row"
      >
        <div
          className="col-md-12"
        >
          <h4
            className="lead text-center m-b-30 m-t-20"
          >
            Where do you want to show notifications?
          </h4>
        </div>
      </div>
      <div
        className="row"
      >
        <div
          className="col-md-1"
        />
        <div
          className="col-md-11"
        >
          <div
            className="ml-5 pl-4 input-group col-md-8"
          >
            <label
              className="pt-2 pl-1 pr-3 text-muted"
            >
              http://localhost:3000
              /
            </label>
            <input
              aria-describedby="urladd"
              className="form-control txtpageurl"
              onBlur={[Function]}
              onChange={[Function]}
              onKeyUp={[Function]}
              placeholder="eg. /mypage, /register, /products"
              type="text"
              value=""
            />
            <span
              className="input-group-btn col-md-2"
              id="urladd"
            >
              <span
                className="btn btn-custom nav nav-pills waves-light waves-effect addpath-btn pl-5 pr-5"
                onClick={[Function]}
              >
                Add
              </span>
            </span>
          </div>
        </div>
      </div>
      <div
        className="pt-2 path-error row"
      >
        <span
          className="help-block"
        >
          <p
            className="website-error"
          >
            
          </p>
        </span>
      </div>
      <div
        className="row"
      >
        <div
          className="col-md-12"
        >
          <div
            className="status"
          >
            <ul>
              <li>
                <span
                  className="dot success"
                />
                <span
                  className="status-name"
                >
                  Active
                </span>
              </li>
              <li>
                <span
                  className="dot primary"
                />
                <span
                  className="status-name"
                >
                  Last seen over 24 hrs ago
                </span>
              </li>
              <li>
                <span
                  className="dot unverified"
                />
                <span
                  className="status-name"
                >
                  Has Never Been Tracked
                </span>
              </li>
              <li>
                <span
                  className="dot danger"
                />
                <span
                  className="status-name"
                >
                  Invalid URL
                </span>
              </li>
            </ul>
          </div>
        </div>
      </div>
      <div
        className="row"
      >
        <div
          className="col-md-12"
        >
          <div
            className="card card-plain"
          >
            <div
              className="content"
            >
              <div
                style={
                  Object {
                    "height": 300,
                    "overflow": "hidden",
                    "position": "relative",
                    "width": "100%",
                  }
                }
              >
                <div
                  style={
                    Object {
                      "WebkitOverflowScrolling": "touch",
                      "bottom": 0,
                      "left": 0,
                      "marginBottom": 0,
                      "marginRight": 0,
                      "overflow": "scroll",
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                    }
                  }
                >
                  <div
                    className="centertbl"
                  >
                    <table
                      className="table"
                    >
                      <thead>
                        <tr>
                          <th>
                            PATH
                          </th>
                          <th>
                            STATUS
                          </th>
                          <th>
                            TRASH
                          </th>
                        </tr>
                      </thead>
                      <tbody />
                    </table>
                  </div>
                </div>
                <div
                  style={
                    Object {
                      "borderRadius": 3,
                      "bottom": 2,
                      "display": "none",
                      "height": 6,
                      "left": 2,
                      "position": "absolute",
                      "right": 2,
                    }
                  }
                >
                  <div
                    style={
                      Object {
                        "backgroundColor": "rgba(0,0,0,.2)",
                        "borderRadius": "inherit",
                        "cursor": "pointer",
                        "display": "block",
                        "height": "100%",
                        "position": "relative",
                      }
                    }
                  />
                </div>
                <div
                  style={
                    Object {
                      "borderRadius": 3,
                      "bottom": 2,
                      "display": "none",
                      "position": "absolute",
                      "right": 2,
                      "top": 2,
                      "width": 6,
                    }
                  }
                >
                  <div
                    style={
                      Object {
                        "backgroundColor": "rgba(0,0,0,.2)",
                        "borderRadius": "inherit",
                        "cursor": "pointer",
                        "display": "block",
                        "position": "relative",
                        "width": "100%",
                      }
                    }
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="float-left"
      >
        <button
          className="btn btn-primary waves-effect number displaybtn-back"
          onClick={[Function]}
          type="button"
        >
          <i
            className="icon-arrow-left pr-2"
          />
          Back
        </button>
      </div>
      <div
        className="float-right"
      >
        <button
          className="btn btn-primary  waves-effect number pl-4 pr-3 displaybtn-finish"
          onClick={[Function]}
          type="button"
        >
          Next
          <i
            className="icon-arrow-right pl-2"
          />
           
        </button>
      </div>
      <div
        className="clearfix"
      />
    </div>
  </div>
</div>
`;
